src/solve.jl

```diff
@@ -77,197 +102,110 @@ A solution object whose content depends on the input options.
 To see all available input options, see
 `keys(Reachability.available_keywords.dict)`.
 """
-function solve(system::InitialValueProblem,
-               options_input::Options)::AbstractSolution
-
-    # ==========
-    # Dimensions
-    # ==========
-    options_input.dict[:n] = statedim(system)
-
-    # =======
-    # Options
-    # =======
-    # solver-specific options (adds default values for unspecified options)
-    options = validate_solver_options_and_add_default_values!(options_input)
-
-    # ===================
-    # Time discretization
-    # ===================
-    if system isa InitialValueProblem{<:AbstractContinuousSystem}
-        info("Time discretization...")
-        tic()
-        Δ = discretize(
-            system,
-            options[:δ],
-            approx_model=options[:approx_model],
-            pade_expm=options[:pade_expm],
-            lazy_expm=options[:lazy_expm_discretize],
-            lazy_sih=options[:lazy_sih]
-            )
-        tocc()
-    else
-        Δ = system
-    end
+function solve!(system::InitialValueProblem, options::Options; algorithm::String=default_algorithm(system))::AbstractSolution
 
-    # ==============
-    # Transformation
-    # ==============
-    if options[:coordinate_transformation] != ""
-        info("Transformation...")
-        tic()
-        (Δ, transformation_matrix) = transform(Δ, options[:coordinate_transformation])
-        tocc()
-    else
-        transformation_matrix = nothing
+#=
+    if :algorithm ∉ keys(options)
+        options[:algorithm] = default_algorithm(system)
     end
+=#
+    println(algorithm)
```

Comment:
You have to decide between `algorithm` and `options[:algorithm]`. Only use one of them.